apps/passport-client/src/validateState.ts

```diff
@@ -196,6 +220,20 @@ export function getRunningAppStateValidationErrors(
     // places that the user's commitment exists match - in the self, the
     // identity, and in the pcd collection
 
+    if (publicKeyV4FromSelfField !== publicKeyV4FromIdentityField) {
+      errors.push(
+        `public key in 'self' field of app state (${publicKeyV4FromSelfField})` +
+          ` does not match public key of 'identityV4' field of app state (${publicKeyV4FromIdentityField})`
+      );
+    }
+
+    if (commitmentV4FromSelfField !== commitmentV4FromIdentityField) {
```

Comment:
The if before this block seems to be about V3 only, so maybe these V4 clauses belong below?
Also I think this clause is duplicated identically below.

Reply:
This code has been substantially changed since this comment, so I'm resolving it to make space for new comments.